### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by artemd24**

### CODE REVIEW

File: homework5/tests/task1_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from homework5.oop_1 import Homework, Student, Teacher
+
+
+def test_teacher_creation():
+    teacher = Teacher("Daniil", "Shadrin")
+    assert teacher.__getattribute__("last_name") == "Daniil"
+    assert teacher.__getattribute__("first_name") == "Shadrin"
```
Comment (apaniukov) [inline homework5/tests/task1_test.py]: You could use a simple `instance.attribute` instead of calling magic methods.

Comment (ArtemDavletov) [reply homework5/tests/task1_test.py:7]: I write `getattr(student, "last_name")`

Comment (artemd24) [reply homework5/tests/task1_test.py]: Done

File: homework5/tests/task2_test.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    decorated_custom_sum = print_result(custom_sum)
+
+    assert decorated_custom_sum.__name__ == custom_sum.__name__ == "custom_sum"
+
+
+def test_custom_sum_original_func():
+    custom_sum = set_custom_sum_mock()
+    decorated_custom_sum = print_result(custom_sum)
+
+    assert decorated_custom_sum.__original_func() == custom_sum() == 10
```
Comment (apaniukov) [inline homework5/tests/task2_test.py:31]: You should check if the `print_result` decorator actually print something to stdout.

**Review by apaniukov — CHANGES_REQUESTED** [review]
Add stdout check for `print_result` decorator.

**Review by apaniukov — APPROVED** [review]

**Pull request merged**